api/pkg/handler/node_v1.go

```
@@ -875,20 +881,140 @@ func getNodeDeploymentForCluster(projectProvider provider.ProjectProvider) endpo
+		}
+
+		// We cannot use machineClient.ClusterV1alpha1().MachineDeployments().Patch() method as we are not exposing
+		// MachineDeployment type directly. API uses NodeDeployment type and we cannot ensure compatibility here.
+		machineDeployment, err := machineClient.ClusterV1alpha1().MachineDeployments(metav1.NamespaceSystem).Get(req.NodeDeploymentID, metav1.GetOptions{})
+		if err != nil {
+			return nil, kubernetesErrorToHTTPError(err)
+		}
+
+		nodeDeployment, err := outputMachineDeployment(machineDeployment)
+		if err != nil {
+			return nil, fmt.Errorf("cannot output existing node deployment: %v", err)
+		}
+
+		nodeDeploymentJSON, err := json.Marshal(nodeDeployment)
+		if err != nil {
+			return nil, fmt.Errorf("cannot decode existing node deployment: %v", err)
+		}
+
+		patchedNodeDeploymentJSON, err := jsonpatch.MergePatch(nodeDeploymentJSON, req.Patch)
+		if err != nil {
+			return nil, fmt.Errorf("cannot patch node deployment: %v", err)
+		}
+
+		var patchedNodeDeployment *apiv1.NodeDeployment
+		err = json.Unmarshal(patchedNodeDeploymentJSON, &patchedNodeDeployment)
+		if err != nil {
+			return nil, fmt.Errorf("cannot decode patched cluster: %v", err)
+		}
+
+		dc, found := dcs[cluster.Spec.Cloud.DatacenterName]
+		if !found {
+			return nil, fmt.Errorf("unknown cluster datacenter %s", cluster.Spec.Cloud.DatacenterName)
+		}
+
+		keys, err := sshKeyProvider.List(project, &provider.SSHKeyListOptions{ClusterName: req.ClusterID})
+		if err != nil {
+			return nil, kubernetesErrorToHTTPError(err)
+		}
+
+		patchedMachineDeployment, err := machineresource.Deployment(cluster, patchedNodeDeployment, dc, keys)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create machine deployment from template: %v", err)
+		}
+
+		// Only spec will be updated by a patch.
+		// It ensures that the name and resource version are set and the selector stays the same.
+		machineDeployment.Spec = patchedMachineDeployment.Spec
```

I think that detecting updates to other fields than `spec` can be difficult ? If so, I would appreciate a comment (swagger spec) that explains which fields can be updated. If not we could return an error.

---

Yep, it would be difficult. I will update comments for swagger.